spanner/spanner_samples.rb

```diff
@@ -1495,6 +1505,211 @@ def write_read_float64_array project_id:, instance_id:, database_id:
+  # [END spanner_restore_backup]
+end
+
+def create_backup_cancel project_id:, instance_id:, database_id:, backup_id:, expire_time:
+  # [START spanner_cancel_backup_create]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+  # backup_id = "Your Spanner backup ID"
+  # expire_time = "Your Spanner backup expiration time"
+
+  require "google/cloud/spanner"
+
+  spanner  = Google::Cloud::Spanner.new project: project_id
+  instance = spanner.instance instance_id
+  database = instance.database database_id
+
+  job = database.create_backup backup_id, expire_time
+
+  puts "Backup operation in progress"
+
+  job.cancel
+  job.wait_until_done!
+
+  backup = instance.backup backup_id
+
+  backup.delete if backup
+
+  puts "#{backup_id} creation job cancelled"
+  # [END spanner_cancel_backup_create]
+end
+
+def list_backup_operations project_id:, instance_id:, database_id:
+  # [START spanner_list_backup_operations]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  spanner  = Google::Cloud::Spanner.new project: project_id
+  instance = spanner.instance instance_id
+
+  jobs = instance.backup_operations filter: "metadata.database:#{database_id} AND metadata.@type:CreateBackupMetadata"
+  jobs.each do |job|
+    if job.error?
+      puts job.error
+    else
+      puts "Backup #{job.backup.backup_id} pending: #{job.progress_percent}% complete"
+    end
+  end
+  # [END spanner_list_backup_operations]
+end
+
+def list_database_operations project_id:, instance_id:
+  # [START spanner_list_database_operations]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+
+  require "google/cloud/spanner"
+
+  spanner  = Google::Cloud::Spanner.new project: project_id
+  instance = spanner.instance instance_id
+
+  jobs = instance.database_operations filter: "metadata.@type:RestoreDatabaseMetadata"
+  jobs.each do |job|
+    if job.error?
+      puts job.error
+    else
+      restore_info = job.database.restore_info
+      puts "Database #{job.database.database_id} restored from backup #{restore_info.backup_info.backup_id} is in #{job.database.state} state"
+    end
+  end
+  # [END spanner_list_database_operations]
+end
+
+def list_backups project_id:, instance_id:
+  # [START spanner_list_backups]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+
+  require "google/cloud/spanner"
+
+  spanner  = Google::Cloud::Spanner.new project: project_id
+  instance = spanner.instance instance_id
+
+  puts "All backups"
+  instance.backups.all.each do |backup|
+    puts backup.backup_id
+  end
+
+  puts "All backups that contains a name \"test_bu_\":"
+  instance.backups(filter: "name:test_bu_").all.each do |backup|
```

Comment:
For other languages, we ended up passing in the backup_id to this function and filtering on that backup_id. If you can change it to be the same here, that would be great.

Reply:
done